helix-term/src/commands.rs

```diff
@@ -1,3 +1,8 @@
+// TODO: remove this when the MSRV is 1.62.0.
+// This warning is benign and is removed in 1.62.0.
+// See https://github.com/rust-lang/rust/pull/96268
+#![allow(mutable_borrow_reservation_conflict)]
```

Comment:
The linked PR has some context on why this warning is removed and why it's ok to ignore.

It allows us to write in a style like so:

```rs
let (view, doc) = current_ref!(cx.editor);
// make some immutable borrows of `view` and `doc`
let transaction = // ...
cx.editor.apply_transaction(&transaction, doc.id(), view.id)
```

The immutable borrow of `Document::id` within the mutable borrow of `Editor::apply_transaction` is flagged as problematic by this warning. We could have a separate line for `let doc_id = doc.id()` but since this warning is going away in 1.62.0 I think we should just ignore it